contrib/validation/validate_lethe.sh

```diff
@@ -0,0 +1,209 @@
+                shift 2  # Skip the flag and its argument
+            else
+                echo "Error: Missing argument for -o|--output."
+                exit 1
+            fi
+            ;;
+        *)
+            echo "Unknown option: $1"
+            exit 1
+            ;;
+    esac
+done
+
+# Check if the output path was provided
+if [[ -z $output_path ]]; then
+    echo "Output path not specified. Use the -o or --output flag to set it."
+    exit 1
+fi
+
+echo "The output path has been set to $output_path"
+verify_or_create_folder "$output_path"
+lethe_path=$(pwd)
+echo "The current path is: $lethe_path"
+echo "Please ensure that the current path is the root folder of lethe"
+
+# Check if magick is available. Otherwise reports will not be generated.
+# If magick is not available, the script will exit with an error status.
+# Check if the application exists
+if ! command -v "magick" &> /dev/null; then
+  echo "Error: magick is not installed or not in the PATH." >&2
+  exit 1
+fi
+
+# Check if git is available. Otherwise it is not possible to have a git hash
+if ! command -v "git" &> /dev/null; then
+  error exit echo "Error: git is not installed or not in the PATH." >&2
+  exit 1
+fi
+
+# Check if python3 is available. Otherwise it is not possible to post-process the simulations
+if ! command -v "python3" &> /dev/null; then
+  error exit echo "Error: python3 is not installed or not in the PATH." >&2
+  exit 1
+fi
+
+# Output file
+hash_file="$output_path/current_git_hash.txt"
+
+# Get the current Git hash
+if git rev-parse --is-inside-work-tree > /dev/null 2>&1; then
+    git_hash=$(git rev-parse HEAD)
+    echo "Current Git hash: $git_hash"
+    echo "$git_hash" > "$hash_file"
+    echo "Git hash saved to '$hash_file'."
+else
+    echo "Error: Not inside a Git repository."
+    exit 1
+fi
+
+echo "Press Enter to continue..."
+read -r  # Wait for user to press Enter
+
+
+
+# Input file
+input_file="$lethe_path/contrib/validation/validation_cases.txt"
+
+cases=()
+n_procs=()
+
+# Loop through each line of the file and store the cases and number of procs
+while IFS= read -r line; do
+
+  # Skip lines that are empty or start with a comment
+  if [[ -z "$line" || "$line" =~ ^# ]]; then
+      continue
+  fi
+
+  # Split the line into two parts
+  part1=$(echo "$line" | awk '{print $1}')
+  part2=$(echo "$line" | awk '{print $2}')
+  
+  # Append the parts to the respective arrays
+  cases+=("$part1")
+  n_procs+=("$part2")
+
+done < "$input_file"
+
+#Create the first page of the PDF report
+echo -e "Lethe validation \nDate: $(date) \nGit hash: $git_hash  \nComputer: $(hostname -f) \n" | \
+magick -density 300 -pointsize 12 text:- $output_path/report.pdf
+
+
+echo "Proceeding with validation..."
+echo "-----------------------------"
+date +"%Y-%m-%d: %H:%M"
+for i in "${!cases[@]}"; do
+  echo "---> Processing ${cases[i]} using ${n_procs[i]} cores"
+  cd $lethe_path/${cases[i]}
+  bash validate.sh -p $output_path -np ${n_procs[i]}
```

Comment:
I would say in the show_help that the output_path should be an absolute path. I tried to do something like : "-o ./test" from my Lethe top folder and it broke because of the "cd" command at line 203.

Reply:
Good point